ArcGISRuntimeSDKQt_QMLSamples/EditData/EditFeaturesWithFeatureLinkedAnnotation/EditFeaturesWithFeatureLinkedAnnotation.qml

```diff
@@ -0,0 +1,301 @@
+
+                        if (!selectedFeature)
+                            return;
+
+                        // update the two attirbutes with the inputed text.
+                        selectedFeature.attributes.replaceAttribute("AD_ADDRESS", attAddressTextField.text);
+                        selectedFeature.attributes.replaceAttribute("ST_STR_NAM", attStreetTextField.text);
+                        selectedFeature.featureTable.updateFeature(selectedFeature);
+                    }
+                }
+
+                Button {
+                    Layout.alignment: Qt.AlignRight
+                    Layout.margins: 5
+                    text: qsTr("Cancel")
+                    // once the cancel button is clicked, hide the window and reset text fields
+                    onClicked: {
+                        updateWindow.visible = false;
+                        streetNameText = "";
+                        addressText = "";
+                    }
+                }
+            }
+        }
+    }
+
+    function clearSelection() {
+        pointsFeatureLayer.clearSelection();
+        linesFeatureLayer.clearSelection();
+
+        selectedFeature = null;
+    }
+
+    function moveFeature(mapPoint) {
+        const geometry = selectedFeature.geometry;
+        const workingPoint = GeometryEngine.project(mapPoint, geometry.spatialReference);
+
+        if (geometry.geometryType === Enums.GeometryTypePolyline) {
+
+            // get nearest vertex to the map point on the selected polyline
+            const nearestVertex = GeometryEngine.nearestVertex(geometry, workingPoint);
+
+
+            const polylineBuilder = ArcGISRuntimeEnvironment.createObject("PolylineBuilder", {
+                                                                              geometry: geometry,
+                                                                              spatialReference: Factory.SpatialReference.createWgs84()
+                                                                          });
+
+            // get part of polyline nearest to map point
+            const part = polylineBuilder.parts.part(nearestVertex.partIndex);
+
+            // remove nearest point to map point
+            part.removePoint(nearestVertex.pointIndex);
+
+            // add the map point as a new point
+            part.addPoint(workingPoint);
+
+            // update the selected feature with the new geometry
+            selectedFeature.geometry = polylineBuilder.geometry;
+            selectedFeature.featureTable.updateFeature(selectedFeature);
+            clearSelection();
+        } else if ( geometry.geometryType === Enums.GeometryTypePoint ) {
+
+            // if the selected feature is a point, change the geometry with the map point
+            selectedFeature.geometry = mapPoint;
+
+            // update the selected feature with the new geometry
+            selectedFeature.featureTable.updateFeature(selectedFeature);
+            clearSelection();
```

Comment:
These two lines are common for both cases. Consider moving it out of these blocks to reduce duplication.